domovoi.py

```diff
@@ -58,7 +58,7 @@ def run(self, processes_file):
         self.check_paths(solar_car_processes)
         self.start_processes(solar_car_processes)
         # Watch over the processes and respond accordingly
-        while True:
+        while len(solar_car_processes):
```

Comment:
The intent here is not clear. There should be an explicit separate check, or at least add a comment here that domovoi will exit with no processes in the list.